spec/controllers/books_controller_spec.rb

```diff
@@ -0,0 +1,161 @@
+require 'rails_helper'
+require_relative 'shared_context'
+
+describe BooksController do
+  it { is_expected.to route(:get, '/books').to(action: :index) }
+  it { is_expected.to route(:get, '/books/1').to(action: :show, id: 1) }
+
+  describe 'GET #index' do
+    context 'When an user is authenticated' do
+      include_context 'Authenticated User'
+      let!(:books) { create_list(:book, 5) }
+      context 'When fetching all the books' do
+        before { get :index }
+
+        it { is_expected.to respond_with :ok }
+
+        it 'responses with a json' do
+          expect(response.content_type).to eq 'application/json'
+        end
+
+        it 'responses with a paginated json' do
+          expect(response_body).to be_paginated
+        end
+
+        it 'responses with the books json' do
+          expected = ActiveModel::Serializer::CollectionSerializer.new(
+            books, serializer: BookPreviewSerializer
+          ).to_json
+          expect(response_body['page']).to eq JSON.parse(expected)
+        end
+      end
+
+      context 'When params[:limit] == 2' do
+        before { get :index, params: { limit: 2 } }
+
+        it { is_expected.to respond_with :ok }
+
+        it 'responses with a json' do
+          expect(response.content_type).to eq 'application/json'
+        end
+
+        it 'responses with a paginated json' do
+          expect(response_body).to be_paginated
+        end
+
+        it 'responses with a 2 books json' do
+          expect(response_body['page'].length).to eq 2
+        end
+      end
+
+      context 'When params[:genre] == Action' do
+        before do
+          create_list(:book, 5, genre: 'Action')
+          create_list(:book, 5, genre: 'Fantasy')
+          get :index, params: { genre: 'Action' }
+        end
+        subject { response_body }
+
+        it { is_expected.to be_paginated }
+
+        it 'responses with a json with 5 books' do
+          expect(response_body['page'].length).to eq 5
+        end
+        it 'returns only books with genre == Action' do
+          expect(response_body['page'].map { |book| book['genre'] }).to satisfy do |genres|
+            genres.all? { |genre| genre == 'Action' }
+          end
+        end
+      end
+
+      context 'When params[:author] == Bradford Nolan DDS' do
+        before do
+          create_list(:book, 2, author: 'Bradford Nolan DDS')
+          create_list(:book, 1, author: 'Luis Miguel')
+          get :index, params: { author: 'Bradford Nolan DDS' }
+        end
+        subject { response_body }
+
+        it { is_expected.to be_paginated }
+        it 'responses with a json with 2 books' do
+          expect(response_body['page'].length).to eq 2
+        end
+        it 'returns only books with author == Bradford Nolan DDS' do
```

Comment:
```rb
 it 'returns books filtered by author name' do
authors = response_body['page'].pluck('author')
expect(authors).to(be_all { |author| author.eq(author_name) } )
end
```

Reply:
Thanks! pluck :heart_eyes: 
Is there any reason to use eq instead of == ?